barman/recovery_executor.py

```diff
@@ -1807,13 +1810,321 @@ def _backup_copy(self, backup_info, dest, remote_command=None, **kwargs):
             raise DataTransferFailure.from_command_error("rsync", e, msg)
 
 
+class IncrementalRecoveryExecutor(RemoteConfigRecoveryExecutor):
+    """
+    Recovery executor for recovery of Postgres incremental backups.
+
+    This class implements the combine backup process as well as the
+    recovery of the newly combined backup by reusing some of the logic
+    from the RecoveryExecutor class.
+    """
+
+    def __init__(self, backup_manager):
+        """
+        Constructor
+
+        :param barman.backup.BackupManager backup_manager: the BackupManager
+            owner of the executor
+        """
+        super(IncrementalRecoveryExecutor, self).__init__(backup_manager)
+        self.combine_start_time = None
+        self.combine_end_time = None
+
+    def recover(self, backup_info, dest, remote_command=None, **kwargs):
+        """
+        Performs the recovery of an incremental backup.
+
+        It first combines all backups in the backup chain, full to incremental,
+        then proceeds with the recovery of the generated synthetic backup.
+
+        This method should be called in a closing context.
+
+        :param barman.infofile.BackupInfo backup_info: the backup to recover
+        :param str dest: the destination directory
+        :param str|None remote_command: The remote command to recover
+            the base backup, in case of remote backup.
+        """
+        # First combine the backups, generating a new synthetic backup in the staging area
+        combine_directory = self.config.recovery_staging_path
+        synthetic_backup_info = self._combine_backups(backup_info, combine_directory)
+
+        # Add the backup directory created in the staging area to be deleted after recovery
+        synthetic_backup_dir = synthetic_backup_info.get_basebackup_directory()
+        self.temp_dirs.append(fs.LocalLibPathDeletionCommand(synthetic_backup_dir))
+
+        # Perform the standard recovery process passing the synthetic backup
+        return super(IncrementalRecoveryExecutor, self).recover(
+            synthetic_backup_info, dest, remote_command=remote_command, **kwargs
+        )
+
+    def _combine_backups(self, backup_info, dest):
+        """
+        Combines the backup chain into a single synthetic backup using the
+        pg_combinebackup utility.
+
+        :param barman.infofile.LocalBackupInfo backup_info: an incremental
+            backup in the chain
+        :param str dest: the directory where the synthetic backup is going
+            to be mounted on
+        :return barman.infofile.SyntheticBackupInfo: the backup info file of the
+            combined backup
+        """
+        self.combine_start_time = datetime.datetime.now()
+
+        dest = parse_recovery_staging_path(dest)
```

Comment:
this is checking for absolute path!

Reply:
But I don't think we need here again. This check is done when the user changes the server paramater config and also when he passes it via CLI, so there's basically no way we could have an relative path here, that's why a removed it earilier

Reply:
Oh i did not check the cli! Ok, no problem!